File: typescript/src/errors/ReplayError.ts
import {
  ErrorCode,
  ParmanaError,
} from "./ParmanaError.js";

/**
 * Raised when deterministic replay fails.
 */
export class ReplayError extends ParmanaError {
  constructor(
    message: string,
    options?: {
      requestId?: string;
      cause?: unknown;
    },
  ) {
    super({
      code: ErrorCode.REPLAY_ERROR,

      message,

      ...(options?.requestId !== undefined && {
        requestId: options.requestId,
      }),

      ...(options?.cause !== undefined && {
        cause: options.cause,
      }),
    });

    this.name = "ReplayError";
  }
}

File: python/parmana/errors/replay_error.py
from __future__ import annotations

from .api_error import ParmanaError


class ReplayError(ParmanaError):
    """
    Raised when deterministic replay fails. Mirrors
    typescript/src/errors/ReplayError.ts.
    """

    def __init__(
        self,
        message: str,
        *,
        request_id: str | None = None,
    ) -> None:
        super().__init__(
            message,
            code="REPLAY_ERROR",
            request_id=request_id,
        )
